packages/loot-design/src/components/modals/SelectLinkedAccounts.js

```diff
@@ -1,254 +1,197 @@
-import React, { useEffect, useState } from 'react';
+import React, { useState } from 'react';
 
-import { View, Modal, P, Button, Strong, CustomSelect } from '../common';
+import { colors } from '../../style';
+import { View, Modal, Button, Text } from '../common';
+import Autocomplete from '../NewAutocomplete';
+import { TableHeader, Table, Row, Field } from '../table';
 
-function EmptyMessage() {
-  return null;
-}
+const addAccountOption = { value: 'new', label: 'Create new account' };
 
 export default function SelectLinkedAccounts({
-  upgradingAccountId,
   modalProps,
   requisitionId,
-  accounts: importedAccounts,
-  actualAccounts,
+  externalAccounts,
+  localAccounts,
   actions,
 }) {
-  let [chosenAccounts, setChosenAccounts] = useState([]);
+  const [chosenAccounts, setChosenAccounts] = useState({});
 
-  const addAccountOption = { id: 'new', name: 'Create new account' };
+  async function onNext() {
+    Object.entries(chosenAccounts).forEach(
+      ([chosenExternalAccountId, chosenLocalAccountId]) => {
+        const externalAccount = externalAccounts.find(
+          account => account.account_id === chosenExternalAccountId,
+        );
+
+        actions.linkAccount(
+          requisitionId,
+          externalAccount,
+          chosenLocalAccountId !== addAccountOption.value
+            ? chosenLocalAccountId
+            : undefined,
+        );
+      },
+    );
+
+    actions.closeModal();
+  }
 
-  const importedAccountsToSelect = importedAccounts.filter(
+  const unlinkedAccounts = localAccounts.filter(
     account =>
-      !chosenAccounts
-        .map(acc => acc.chosenImportedAccountId)
-        .includes(account.account_id),
+      !account.account_id &&
+      !Object.values(chosenAccounts).includes(account.id),
   );
 
-  const actualAccountsToSelect = [
-    addAccountOption,
-    ...actualAccounts.filter(
-      account =>
-        !chosenAccounts
-          .map(acc => acc.chosenActualAccountId)
-          .includes(account.id),
-    ),
-  ];
-
-  useEffect(() => {
-    const chosenAccountsToAdd = [];
-    importedAccountsToSelect.forEach(importedAccount => {
-      // Try to auto-match accounts based on account_id or mask
-      // Add matched accounts to list of selected accounts
-      const matchedActualAccount = actualAccountsToSelect.find(
-        actualAccount => {
-          return (
-            actualAccount.account_id === importedAccount.account_id ||
-            actualAccount.mask === importedAccount.mask
-          );
-        },
-      );
-
-      if (matchedActualAccount) {
-        chosenAccountsToAdd.push({
-          chosenImportedAccountId: importedAccount.account_id,
-          chosenActualAccountId: matchedActualAccount.id,
-        });
-      }
-    });
-
-    setChosenAccounts([...chosenAccounts, ...chosenAccountsToAdd]);
-  }, []);
```

Comment:
Do we have anywhere this logic? It's very useful when you re-linking accounts. Thanks to that the list of accounts will be auto-filled for the user.

Reply:
Thanks! For the past few days I've been trying to get the bank-sync expiration to kick in, but I've not been able to get it working.. I tried setting `accessValidForDays` to 1, but that did not seem to make it work.

How did you test this when you did the initial implementation?